### PR TITLE
add Kubernetes config provider

### PR DESCRIPTION
**Pull request by marratj**
This PR adds a Kubernetes config provider that allows infrared to automatically pick up new Kubernetes services and configure proxies for them, similar to how mc-router does it.

It works basically the same as the already existing Docker config provider just that it looks for annotations in Kubernetes Service objects instead of Docker container labels.

### CODE REVIEW

File: internal/pkg/config/provider/k8s.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+				Config: cfg,
+			}
+		}
+
+	}
+}
+
+func (p Kubernetes) Close() error {
+	return nil
+}
```
Comment (haveachin) [inline internal/pkg/config/provider/k8s.go:164]: This is meant for cleanup. Don't you need to close the Clientset?

**Review by haveachin — CHANGES_REQUESTED** [review]
Hey, thanks for the PR! I don't personally use Kubernetes in my setup (for now), so I have to trust you on this to be working. The PR is great, but there are a few points that need some changes/further explanation.
I'm not sure why you are using errors.Wrap instead of fmt.Errorf("error msg: %w", err) to wrap an error. Are there any upsides? What does this library do that the standard library doesn't? Errors should also follow the convention of being all in lower case.

File: internal/pkg/config/provider/k8s.go (hunk 1 above)
Comment (marratj) [reply internal/pkg/config/provider/k8s.go:164]: The `errors.Wrap` actually came from a boilerplate template that I usually use for the Kubernetes connection. I changed it to `fmt.Errorf()` now.

For the Kubernetes Clientset, there actually is no Close (or any other cleanup method) implemented.

**Review by haveachin — CHANGES_REQUESTED** [review]
Looks good! But please also remove the unused dependency from the go.mod.

File: go.mod [hunk 2]
```diff
@@ -22,6 +22,7 @@ require (
 	github.com/klauspost/compress v1.16.5
 	github.com/mitchellh/mapstructure v1.5.0
 	github.com/pires/go-proxyproto v0.7.0
+	github.com/pkg/errors v0.9.1
```
[the remark this answers is not in the document]
Comment (marratj) [reply go.mod]: Done. Just forgot to run `go mod tidy` after I removed it from the code...

**Review by haveachin — APPROVED** [review]
LGTM

**Pull request merged**